test/api.spec.ts

```diff
@@ -19,74 +19,199 @@ import * as bundle from '../src';
 import * as chai from 'chai';
 import * as chaiAsPromised from 'chai-as-promised';
 import { describe } from 'mocha';
-import * as stream from 'stream';
+import { createReadStream, createWriteStream } from 'fs';
+import type { Readable } from 'stream';
 import * as fs from 'fs/promises';
 import * as nock from 'nock';
 import * as resourceBundle from '@balena/resource-bundle';
-import type { ReleaseManifest } from '../src';
 import authToken from './fixtures/authToken';
 import mockResponses from './mocks';
-// import * as path from 'path';
-// import * as SDK from 'balena-sdk';
+import * as path from 'path';
+import * as SDK from 'balena-sdk';
 
 chai.use(chaiAsPromised);
 const expect = chai.expect;
 
 const apiUrl = 'https://api.balena-cloud.com';
 
+async function writeStreamToTarFile(
+	stream: Readable,
+	tarFilePath: string,
+): Promise<void> {
+	const writable = createWriteStream(tarFilePath);
+
+	return new Promise((resolve, reject) => {
+		stream.pipe(writable);
+
+		stream.on('end', resolve);
+		stream.on('error', reject);
+		writable.on('error', reject);
+	});
```

Comment:
“stream/promises” has a “pipeline” function that will do all this (and more) for you and it’s the preferred way to connect streams.